lib/msal-common/src/client/AuthorizationCodeClient.ts

```diff
@@ -121,19 +121,30 @@ export class AuthorizationCodeClient extends BaseClient {
         if (!logoutRequest) {
             throw ClientConfigurationError.createEmptyLogoutRequestError();
         }
+        const queryString = this.createLogoutUrlQueryString(logoutRequest);
+
+        // Construct logout URI.
+        return UrlString.appendQueryString(this.authority.endSessionEndpoint, queryString);
+    }
 
+    /**
+     * Clears cache items belonging to the account provided. If no account is provided, clears all MSAL-generated cache items.
+     * @param logoutRequest 
+     */
+    async clearCacheOnLogout(logoutRequest: CommonEndSessionRequest): Promise<void> {
         if (logoutRequest.account) {
             // Clear given account.
-            this.cacheManager.removeAccount(AccountEntity.generateAccountCacheKey(logoutRequest.account));
+            try {
+                await this.cacheManager.removeAccount(AccountEntity.generateAccountCacheKey(logoutRequest.account));
+                this.logger.verbose("Cleared cache items belonging to the account provided in the logout request.");
+            } catch (error) {
+                this.logger.verbose("Account provided in logout request was not found.");
+            }
         } else {
             // Clear all accounts and tokens
-            this.cacheManager.clear();
+            this.logger.verbose("No account provided in logout request, clearing all cache items.");
+            await this.cacheManager.clear();
```

Comment:
Also curious to know what should happen if this throws

Reply:
I'd be interested in your opinion on what we should do if it throws besides try/catching the error and logging that cache clearing failed. We can proceed with the ending the session with the server even if the cache isn't clear, but the user will still be functionally logged in (having credentials in the cache).

Reply:
Yeah, catching and logging is fine. Would prefer local cache potentially being out of date then the app being stuck.